src/spec-configuration/containerFeaturesOCIPush.ts

```diff
@@ -157,15 +157,27 @@ export async function putManifestWithTags(output: Log, manifestStr: string, feat
 	for await (const tag of tags) {
 		const url = `https://${featureRef.registry}/v2/${featureRef.path}/manifests/${tag}`;
 		output.write(`PUT -> '${url}'`, LogLevel.Trace);
-		const { statusCode, resHeaders } = await requestResolveHeaders({
+
+		const options = {
 			type: 'PUT',
 			url,
 			headers: {
 				'Authorization': `Bearer ${registryAuthToken}`,
 				'Content-Type': 'application/vnd.oci.image.manifest.v1+json',
 			},
 			data: Buffer.from(manifestStr),
-		});
+		};
+
+		let { statusCode, resHeaders } = await requestResolveHeaders(options);
+
+		// Retry logic: when request fails with HTTP 429: too many requests
+		if (statusCode === 429) {
+			output.write(`Failed to PUT manifest for tag ${tag} due to too many requests. Retrying...`, LogLevel.Warning);
+			let response = await requestResolveHeaders(options);
```

Comment:
It would be cool to make in retries to the actual HTTP methods (along with some backoff time).  

Not sure if this will help since it will fire right away, but might be a good mitigation for now

Reply:
I think this change would be more effective if you add an artificial delay here (1 or 2 seconds?).

Reply:
Added a 2s delay